docs/syz_verifier.md

```diff
@@ -0,0 +1,91 @@
+# syz-verifier
+
+Testing and fuzzing aim to verify a subset of the behaviour of running programs: any logical error can cause
+bugs that make the program behave incorrectly, possibly producing unintended output. Many bugs, however, do
+not cause the program to crash nor produce assertion failures. Here we need to distinguish between undefined
+and defined behaviour.
+
+Undefined behaviour can easily be detected by either static (e.g. compiler warnings) or dynamic analysis
+(e.g. sanitizers). However, logical errors that result in defined program behaviour, yet are still causing
+bugs that make the program behave incorrectly (according to some documented or otherwise expected intent),
+are much harder to detect. One way to detect such bugs is setting up test cases that compare program outputs
+against a set of expected results (also called differential fuzzing).
```

Comment:
I wasn't thinking of differential fuzzing yet and I wanted you to keep the original paragraph where you explained differential fuzzing. Sorry, I should have been clearer about that. Setting up test cases can take many forms, but I was thinking of manual things like unit testing at the end here.

If you want you can make my original sentence clearer by adding something like "(for example by manually creating integration or unit tests).". Then the next paragraph is like the one you had explaining what differential fuzzing is, perhaps suggesting it's more automated "Differential fuzzing automates...".

The main thing is to provide contrast and highlight differences, as that'll be able to help people categorize the whole thing better (similar to what I tried to do contrasting undefined vs. defined behavior and where static+dynamic analysis fit there).